File: w1_p4/20.c
#include <stdio.h>
#include <string.h>

int main() {
	int m, n, c=1;
	
	while(scanf("%d %d", &m, &n) == 2){
		if(m==0 && n==0) break;
		getchar();
		char input[102][102];
		int map[102][102] = {0};
		for(int i=1; i<m+1; i++){
			for(int j=1; j<n+1; j++){
				scanf("%c", &input[i][j]);
				if(input[i][j] == '*')
					map[i][j] = -100;
				else
					map[i][j] = 0;
			}
			getchar();
		}
		for(int i=0; i<m+2; i++){
			for(int j=0; j<n+2; j++){
				if(map[i][j] < 0){
					for(int k=i-1; k<=i+1; k++){
						for(int p=j-1; p<=j+1; p++){
							map[k][p] += 1;
						} 
					} 
				}
			}
		}
		printf("Field #%d:\n", c);
		c++;
		for(int i=1; i<m+1; i++){
			for(int j=1; j<n+1; j++)
				if(map[i][j] < 0)
					printf("*");
				else
					printf("%d", map[i][j]);
			printf("\n");
		}
		printf("\n");
	}
	
	return 0;
}

File: w1_p4/20.java
import java.util.Scanner;
class Main {
	public static void main(String[] args) throws Exception {
		Scanner input = new Scanner(System.in);
		int FieldCount = 1;
		
		while(input.hasNextLine()){
			String s = input.nextLine();
			int row=s.charAt(0)-48, col=s.charAt(2)-48;
			if(row != 0 && col != 0) {
				char[][] array1 = new char[row][col];
				char[][] array2 = new char[row][col];
				
				for(int i = 0; i < row; i++){
					s = input.nextLine();
					for(int j = 0; j < col; j++){
						array1[i][j] = s.charAt(j);
					}
				}
				
				System.out.println("Field #"+FieldCount+":");
				for(int i = 0; i < row; i++){
					for(int j = 0; j < col; j++){
						if(array1[i][j] == '*')
							array2[i][j] = '*';
						else{
							int count = 0, kf = i-1, kl = i+1, lf = j-1, ll = j+1;
							if(i-1 < 0)
								kf=i;
							if(i+1 >= row)
								kl=row-1;
							if(j-1 < 0)
								lf=j;
							if(j+1 >= col)
								ll=col-1;
							for(;kf <= kl; kf++){
								for(int g = lf; g <= ll; g++)
									if(array1[kf][g] == '*')
										count++;
								array2[i][j] = Character.forDigit(count, 10);
							}
						}
						System.out.print(array2[i][j]);
					}
					System.out.println();
				}
				FieldCount++;
			}
			System.out.println();
		}
	}
}